packages/fast-components-react-base/src/listbox/listbox.tsx

```diff
@@ -0,0 +1,605 @@
+                        focusItemId,
+                        this.props.children
+                    );
+                    if (itemData !== null) {
+                        this.toggleItem(itemData);
+                    }
+                }
+                break;
+
+            case KeyCodes.end:
+                event.preventDefault();
+                if (this.props.multiselectable && event.shiftKey && event.ctrlKey) {
+                    this.selectRange(
+                        this.state.focusIndex,
+                        this.domChildren().length - 1
+                    );
+                }
+                this.setFocus(this.domChildren().length - 1, -1);
+
+                break;
+
+            case KeyCodes.home:
+                event.preventDefault();
+                if (this.props.multiselectable && event.shiftKey && event.ctrlKey) {
+                    this.selectRange(0, this.state.focusIndex);
+                }
+                this.setFocus(0, 1);
+
+                break;
+
+            default:
+                if (event.key === "A") {
+                    event.preventDefault();
+                    this.selectRange(0, this.domChildren().length);
+                } else if (!event.ctrlKey) {
+                    this.processTypeAhead(event);
+                }
+        }
+    };
+
+    /**
+     * Gets a child node from it's id by examining provided children
+     */
+    private getNodeById = (
+        itemId: string,
+        children: React.ReactNode
+    ): React.ReactNode => {
+        const childrenAsArray: React.ReactNode[] = React.Children.toArray(children);
+
+        const matchNode: React.ReactNode = childrenAsArray.find(
+            (child: React.ReactElement<any>): boolean => {
+                if (
+                    child.props[Listbox.idPropertyKey] === undefined ||
+                    child.props[Listbox.idPropertyKey] !== itemId
+                ) {
+                    return false;
+                }
+                return true;
+            }
+        );
+
+        return matchNode;
+    };
+
+    /**
+     * Sets focus based on characters typed
+     */
+    private processTypeAhead = (e: React.KeyboardEvent<HTMLDivElement>): void => {
+        e.preventDefault();
+
+        clearTimeout(this.typeAheadTimer);
+
+        this.typeAheadString = this.typeAheadString + e.key.toLowerCase();
+
+        let matchIndex: number = -1;
+
+        const children: React.ReactNode[] = React.Children.toArray(this.props.children);
+
+        children.some(
+            (child: React.ReactElement<any>, index: number): boolean => {
+                if (child.props[this.props.typeAheadPropertyKey] === undefined) {
+                    return false;
+                }
+                if (
+                    child.props[this.props.typeAheadPropertyKey]
+                        .toLowerCase()
+                        .startsWith(this.typeAheadString)
```

Comment:
startsWith does not work with IE, any version https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith

Reply:
Presumably the one in @microsoft/fast-web-utilities is good
                 startsWith(
                        child.props[this.props.typeAheadPropertyKey].toLowerCase(),
                        this.typeAheadString
                    )